File: 0007-reverse-integer/0007-reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        long ans = 0;
        while(x){
            int D = x%10;
            ans = ans*10 + D;
            if(ans>INT_MAX || ans<INT_MIN)
                return 0;
            x /= 10;
        }
        return (int)ans;
    }
};

File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        int rev = 0, lastValidNum = Integer.MAX_VALUE / 10;
        boolean negative = x < 0;
        x = Math.abs(x);
        while (x > 0) {
            if (rev > lastValidNum)
                return 0;
            rev *= 10;
            rev += x % 10;
            x /= 10;
        }
        return negative ? -rev : rev;
    }
}